File: php-poo/aulas/aula15/Video.php
<?php
require_once 'AcoesVideo.php';
class Video implements AcoesVideo
{
    // Atributos
    private $titulo;
    private $avaliacao;
    private $views;
    private $curtidas;
    private $reproduzindo;

    /*
     *  class Video implements AcoesVideo (Metodos de implementação)
     *  Vou sobre-escrever os 3 métodos, sendo eles: play(), pause() e like()
     *  @Override (@SobreEscrever)
     *      JAVA => obrigatório declarar antes!
     *      PHP => não declarada nada antes!
     */

    // @Override
    public function play()
    {
        $this->reproduzindo = true;
    }
    public function pause()
    {
        $this->reproduzindo = false;
    }
    public function like()
    {
        $this->curtidas++;
    }


    // Método construtor => recebe somente um parametro que é o titulo
    public function __construct($titulo)
    {
        $this->titulo = $titulo;
        $this->avaliacao = 1;
        $this->views = 0;
        $this->curtidas = 0;
        $this->reproduzindo = false;
    }

    // Métodos especiais, Getters and Setters
    public function getTitulo()
    {
        return $this->titulo;
    }

    public function setTitulo($titulo): void
    {
        $this->titulo = $titulo;
    }

    public function getAvaliacao()
    {
        return $this->avaliacao;
    }

    public function setAvaliacao($avaliacao): void
    {
        $media = ($this->avaliacao + $avaliacao) / ($this->views);
        $this->avaliacao = $media;
    }

    public function getViews()
    {
        return $this->views;
    }

    public function setViews($views): void
    {
        $this->views = $views;
    }

    public function getCurtidas()
    {
        return $this->curtidas;
    }

    public function setCurtidas($curtidas): void
    {
        $this->curtidas = $curtidas;
    }

    public function getReproduzindo()
    {
        return $this->reproduzindo;
    }

    public function setReproduzindo($reproduzindo): void
    {
        $this->reproduzindo = $reproduzindo;
    }

}

File: java-poo/projects/poo-aula14/src/aula14/Video.java
package aula14;

public class Video implements AcoesVideo {
    // Attributes
    private String titulo;
    private int avaliacao;
    private int views;
    private int curtidas;
    private boolean reproduzindo;

    // Constructor
    public Video(String titulo) {
        this.titulo = titulo;
        this.avaliacao = 1;  // estrelas DE 1 ATE 5!
        this.views = 0;
        this.curtidas = 0;
        this.reproduzindo = false;
    }

    // toString()
    @Override
    public String toString() {
        return "Video{" +
                "titulo='" + titulo + '\'' +
                ", avaliacao=" + avaliacao +
                ", views=" + views +
                ", curtidas=" + curtidas +
                ", reproduzindo=" + reproduzindo +
                '}';
    }

    // Special Method - Getters and Setters
    public String getTitulo() {
        return titulo;
    }
    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }
    public int getAvaliacao() {
        return avaliacao;
    }
    public void setAvaliacao(int avaliacao) {
        this.avaliacao = avaliacao;
    }
    public int getViews() {
        return views;
    }
    public void setViews(int views) {
        this.views = views;
    }
    public int getCurtidas() {
        return curtidas;
    }
    public void setCurtidas(int curtidas) {
        this.curtidas = curtidas;
    }

    @Override
    public void play() {
        this.reproduzindo = true;
    }

    @Override
    public void pause() {
        this.reproduzindo = false;
    }

    @Override
    public void like() {
        this.curtidas++;
    }
}
